3rdParty/iresearch/core/search/disjunction.hpp

```diff
@@ -1153,9 +1153,11 @@ class block_disjunction final : public doc_iterator, private score_ctx {
           if constexpr (is_min_match) {
             match_count_ = 1;
           }
-        } else if constexpr (is_min_match) {
-          if (target == value) {
-            ++match_count_;
+        } else {
```

Comment:
Why this change was made? Some compiler bug? constructs like  "else if constexpr" seems to be accepted by gcc, clang and msvc https://godbolt.org/z/fE5Pc9cae

Reply:
The original code caused MSVC to create a warning as indicated:
```c++
      visit_and_purge([this, target, &doc](auto& it) mutable {
        UNUSED(this);
        const auto value = it->seek(target);

        if (doc_limits::eof(value)) {
          // exhausted
          return false;
        }

        // this is to circumvent bug in GCC 10.1 on ARM64
        constexpr bool is_min_match = traits_type::min_match();
        if (value < doc.value) {
          doc.value = value;
          if constexpr (is_min_match) {
            match_count_ = 1;
          }
        } else if constexpr (is_min_match) {
          if (target == value) {
            ++match_count_;
          }
        }

        return true; // warning: not all control paths return a value
      });
```
And it then actually invoked undefined behavior and made the tests fail. This is a compiler bug.

Reply:
And that warning has appeared only after c++20 was enabled ?

Reply:
Just tried to build this on my  fresh-updated msvc - no such  warning for this code 😕  Maybe this is because MSVC on Jenkins is outdated?

Reply:
Yes. Either `/std:c++20` or `/std:c++latest`.

Reply:
Interesting, @mpoeter reproduced it locally and in Jenkins. Maybe we just have to upgrade msvc. 🤔

Reply:
Jenkins has the following version of VS 2019: `16.11.5` (was released 12.10.2021). It's not the latest one (which is `16.11.6` from 07.11.2021).

Reply:
But I've compiled this with "devel" settings e.g. c++17 
I`ll try to compile this branch with  this change reverted and see. Maybe this will change the outcome.

Reply:
Okay, I could confirm that even the latest MSVC with the c++latest have this warning "not all control paths return a value"  (with the original code)